reconcile/terraform_vpc_peerings.py

```diff
@@ -456,6 +456,208 @@ def build_desired_state_vpc(
+              automationToken {
+                path
+                field
+                version
+                format
+              }
+            }
+            vpc_id
+            cidr_block
+            region
+          }
+          assumeRole
+        }
+        ... on ClusterPeeringConnectionAccountVPCMesh_v1 {
+          account {
+            name
+            uid
+            terraformUsername
+            automationToken {
+              path
+              field
+              version
+              format
+            }
+          }
+          tags
+        }
+        ... on ClusterPeeringConnectionAccountTGW_v1 {
+          account {
+            name
+            uid
+            terraformUsername
+            automationToken {
+              path
+              field
+              version
+              format
+            }
+          }
+          tags
+          cidrBlock
+          manageSecurityGroups
+          assumeRole
+        }
+        ... on ClusterPeeringConnectionClusterRequester_v1 {
+          cluster {
+            name
+            network {
+              vpc
+            }
+            spec {
+              region
+            }
+            awsInfrastructureAccess {
+              awsGroup {
+                account {
+                  name
+                  uid
+                  terraformUsername
+                  automationToken {
+                    path
+                    field
+                    version
+                    format
+                  }
+                }
+              }
+              accessLevel
+            }
+            awsInfrastructureManagementAccounts {
+              account {
+                name
+                uid
+                terraformUsername
+                resourcesDefaultRegion
+                automationToken {
+                  path
+                  field
+                  version
+                  format
+                }
+              }
+              accessLevel
+              default
+            }
+
+            peering {
+              connections {
+                name
+                provider
+                manageRoutes
+                ... on ClusterPeeringConnectionClusterAccepter_v1 {
+                  name
+                  cluster {
+                    name
+                  }
+                  awsInfrastructureManagementAccount {
+                    name
+                    uid
+                    terraformUsername
+                    automationToken {
+                      path
+                      field
+                      version
+                      format
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    disable {
+      integrations
+    }
+  }
+}
+"""
+
+
+def fetch_clusters_with_peerings() -> list[dict[str, Any]]:
+    clusters = gql.get_api().query(CLUSTER_QUERY)["clusters"]
```

Comment:
why are you re-defining the `CLUSTER_QUERY` here? Why not use `queries.get_clusters()`?

Reply:
The other query is too big and returns way too many details which has the potential to prevent early exit unnecessarily. 

Does that make sense?

Reply:
Schemasparsamkeit 😜